test/dynamo/test_misc.py

```diff
@@ -1596,6 +1596,34 @@ def fn(x):
             self.assertEqual(ref, res)
         self.assertEqual(cnts.frame_count, 1)
 
+    def test_numpy_tolist(self):
+        def fn(x):
+            return x.tolist()
+
+        cnts = torch._dynamo.testing.CompileCounter()
+        opt_fn = torch._dynamo.optimize(cnts, nopython=True)(fn)
+
+        x = np.arange(5)
+        r = opt_fn(x)
+
+        self.assertEqual(r, [0, 1, 2, 3, 4])
+        self.assertEqual(type(r), list)
+        self.assertEqual(cnts.frame_count, 1)
+
+    def test_numpy_size_attr(self):
+        def fn(x):
+            return x.size + x
```

Comment:
Wait, hwo can this work? Shouldn't it be `x.size()`??

Reply:
That's precisely the point --- in numpy `.size` is an attribute, not a method:

```
In [1]: import numpy as np

In [2]: a = np.arange(9).reshape((3, 3))

In [3]: a.size
Out[3]: 9

In [4]: a.size()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[4], line 1
----> 1 a.size()

TypeError: 'int' object is not callable
```

The fact that dynamo needs a `.size()` method internally is likely an artifact of NumpyNdarrayVariable inheriting from TensorVariable (Liskov? Never heard of him). However as long as it's not visible to users either on eager or under dynamo, it's not more than a dusty corner, not necessarily worth sweeping right now.

Reply:
Yeah, the whole Liskov principle is broken left right and centre in dynamo.
I see that it works because we also implement the property size at
```
        elif name == "size":
            if not free_symbols(r := example_ndarray.size):
                return ConstantVariable.create(int(r), **options)
```

Oh well. Let's leave it as-is for now, yeah.

Reply:
> we also implement the property size 

Precisely the base of the attempt to remove the method call :-). But then --- from eyeballing tracebacks --- somewhere something expects that `FakeTensor` has the `size()` method to implement some not-identical but related numpy API.

Reply:
not for this PR, but could you put in a comment in the code explaining this issue or pointing to this discusison in your next PR?